gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+import torch
+
+from ..utils import cached
+from .added_diag_lazy_tensor import AddedDiagLazyTensor
+from .diag_lazy_tensor import DiagLazyTensor
+from .kronecker_product_lazy_tensor import KroneckerProductLazyTensor
+from .non_lazy_tensor import lazify
+
+# TODO: check the speed of the linear algebra - svd vs eig etc.
+
+
+class KroneckerProductAddedDiagLazyTensor(AddedDiagLazyTensor):
+    def __init__(self, *lazy_tensors):
+        super(KroneckerProductAddedDiagLazyTensor, self).__init__(*lazy_tensors)
+        if len(lazy_tensors) > 2:
+            raise RuntimeError("An AddedDiagLazyTensor can only have two components")
+        elif isinstance(lazy_tensors[0], DiagLazyTensor):
+            self._diag_tensor = lazy_tensors[0]
+            self._lazy_tensor = lazy_tensors[1]
+        elif isinstance(lazy_tensors[1], DiagLazyTensor):
+            self._diag_tensor = lazy_tensors[1]
+            self._lazy_tensor = lazy_tensors[0]
+        else:
+            raise RuntimeError("One of the LazyTensors input to AddedDiagLazyTensor must be a DiagLazyTensor!")
+
+    def inv_quad_logdet(self, inv_quad_rhs=None, logdet=False, reduce_inv_quad=True):
+        if inv_quad_rhs is not None:
+            inv_quad_term = self.inv_quad(inv_quad_rhs)
+        else:
+            inv_quad_term = None
+
+        if logdet is not None:
+            logdet_term = self.logdet()
+        else:
+            logdet_term = None
+
+        return inv_quad_term, logdet_term
+
+    def logdet(self):
+        noisy_eigenvalues = self._kron_eigenvalues.diag() + self._diag_tensor.diag()
```

Comment:
naming nit: "noisy" makes it seem that jitttering is the only thing this is useful for (which it's not).